components/debounce/browser/test/debounce_rule_unittest.cc

```diff
@@ -0,0 +1,409 @@
+  }
+}
+
+TEST(DebounceRuleUnitTest, DeAmpUrl) {
+  TestingPrefServiceSimple prefs;
+  prefs.registry()->RegisterBooleanPref("brave.de_amp.enabled", true);
+
+  const std::string contents = R"json(
+
+      [{
+          "include": [
+              "*://*.cdn.ampproject.org/c/s/*"
+          ],
+          "exclude": [
+          ],
+          "action": "regex-path",
+          "prepend_scheme": "https",
+          "pref": "brave.de_amp.enabled",
+          "param": "^/c/s/(.*)$"
+      }]
+      
+      )json";
+  std::vector<std::unique_ptr<DebounceRule>> rules = StringToRules(contents);
+
+  for (const std::unique_ptr<DebounceRule>& rule : rules) {
+    CheckApplyResult(rule.get(),
+                     GURL("https://www-theverge-com.cdn.ampproject.org/c/s/"
+                          "www.theverge.com/platform/amp/2018/9/20/17881766/"
+                          "bing-google-amp-support-mobile-news"),
+                     "https://www.theverge.com/platform/amp/2018/9/20/17881766/"
+                     "bing-google-amp-support-mobile-news",
+                     &prefs, false);
+  }
+}
+
+TEST(DebounceRuleUnitTest, DotomiTrackerUrl) {
+  const std::string contents = R"json(
+
+      [{
+          "include": [
+              "*://*.dotomi.com/links-t/*"
+          ],
+          "exclude": [
+          ],
+          "action": "regex-path",
+          "param": "^/links-t/[0-9]*/type/dlg/sid/---/(.*)$"
+      }]
+      
+      )json";
+  std::vector<std::unique_ptr<DebounceRule>> rules = StringToRules(contents);
+
+  // From https://github.com/brave/brave-browser/issues/22429
+  auto* tracker_url =
+      "https://cj.dotomi.com/links-t/123/type/dlg/sid/---/https://"
+      "www.carhartt.com/product/123/nintendo-one";
+  auto* target_url = "https://www.carhartt.com/product/123/nintendo-one";
+
+  for (const std::unique_ptr<DebounceRule>& rule : rules) {
+    CheckApplyResult(rule.get(), GURL(tracker_url), target_url, false);
+  }
+}
+
+TEST(DebounceRuleUnitTest, UrlEncoded) {
+  const std::string contents = R"json(
+
+      [{
+          "include": [
+              "*://prf.hn/click/*"
+          ],
+          "exclude": [
+          ],
+          "action": "regex-path",
+          "param": "^/click/camref:[0-9a-zA-Z]*/pubref:[0-9a-zA-Z-]*/destination:(.*)$"
+      }]
+      
+      )json";
+  std::vector<std::unique_ptr<DebounceRule>> rules = StringToRules(contents);
+
+  // From https://github.com/brave/brave-browser/issues/22429
+  auto* tracker_url =
+      "https://prf.hn/click/camref:1011l7xH5/"
+      "pubref:cn-182c2c19daa548769cf89ef10d5a5af3-dtp/"
+      "destination:https%3A%2F%2Fwww.test.com%2Ftest1%2Ftest2";
+  auto* target_url = "https://www.test.com/test1/test2";
+
+  for (const std::unique_ptr<DebounceRule>& rule : rules) {
+    CheckApplyResult(rule.get(), GURL(tracker_url), target_url, false);
+  }
+}
+
+TEST(DebounceRuleUnitTest, AnrdoezrsTrackerUrl) {
```

Comment:
AndroezrsTracker really doesn't have any meaning to me, what is this rule actually testing?

Reply:
Fair